nixos/modules/services/misc/navidrome.nix

```diff
@@ -0,0 +1,196 @@
+        default = "";
+        description = ''
+        Folder where your music libraryis stored. Can be read-only
+        but make sure it is readable.
+        '';
+      };
+
+      musicFolderPermissions = mkOption {
+        type = types.str;
+        default = "775";
+        description = ''
+          The permissions to set for the music folder.
+          They will be applied on every service start.
+        '';
+      };
+      
+      scanInterval = mkOption {
+        type = types.str;
+        default = "1m";
+        description = "How frequently to scan for changes in your music library";
+      };
+
+      logLevel = mkOption {
+        type = types.str;
+        default = "info";
+        description = ''
+        Log level. Useful for troubleshooting.
+        Possible values: error, info, debug, trace
+        '';
+      };
+      
+      dataFolder = mkOption {
+        type = types.path;
+        default = "/var/lib/navidrome";
+        description = "Folder to store application data (DB, cache…)";
+      };
+
+      virtualHost = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+        Name of the nginx virtualhost to use and setup.
+        If null, do not setup any virtualhost.
+        '';
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 4533;
+        description = "HTTP port Navidrome will use";
+      };
+
+      enableTranscodingConfig = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enables transcoding configuration in the UI";
+      };
+
+      transcodingCacheSize = mkOption {
+        type = types.str;
+        default = "100MB";
+        description = "Size of transcoding cache";
+      };
+
+      imageCacheSize = mkOption {
+        type = types.str;
+        default = "100MB";
+        description = "Size of image (art work) cache. set to 0 to disable cache";
+      };
+
+      sessionTimeout = mkOption {
+        type = types.str;
+        default = "24h";
+        description = "How long Navidrome will wait before closing web ui idle sessions";
+      };
+
+      baseUrl = mkOption {
+        type = types.path;
+        default = "/";
+        description = "Base URL (only the path part) to configure Navidrome behind a proxy (ex: /music)";
+      };
+
+      uiLoginBackgroundUrl = mkOption {
+        type = types.str;
+        default = "";
+        description = "Change backaground image used in the Login page";
+      };
+
+      ignoredArticles = mkOption {
+        type = types.str;
+        default = "The El La Los Las Le Les Os As O A";
+        description = "List of ignored articles when sorting/indexing artists";
+      };
+      
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.musicFolder}' '${cfg.musicFolderPermissions}' '${cfg.user}' '${cfg.group}' - -"
+    ];
+    
+    systemd.services.navidrome = {
```

Comment:
It would be a good idea to reuse the upstream's systemd [unit](https://github.com/deluan/navidrome/blob/ad725ac355798a78b1ca47b8bf25f46b1e95d374/contrib/navidrome.service).

You should modify the package to install `navidrome.service` in `$out/lib/systemd/system/`, then to use the unit you can do
```
  systemd.packages = [ pkgs.navidrome ];
```
in the NixOS configuration. Now you only need to set the properties like `environment` and `serviceConfig.User`.

Reply:
> It would be a good idea to reuse the upstream's systemd [unit](https://github.com/deluan/navidrome

@rnhmjoj I understand your idea and approve it, however at the moment the package is built downloading the compiled binary, so I can't copy the systemd config file from the repository.
Obviously I could download both the binary and the repository, but since I was thinking to rewrite to package derivation to build the binary from the sources (as you suggested in another issue) I think to delay this idea for the future.
Is this ok for you?


> Maybe you could create a `settings` option to represent the configuration file which could replace a few of the options you have included in this module. 

@aanderse So do you suggest to fold all the settings inside a single `settings` option? I agree on this, however with the actual derivation every options has a default value which reflects the default value specified in the navidrome documentation. This is hardcoded, maybe an empty default value would be better, so navidrome could choose the default one and this module derivation would not depend on the package version.


Moreover, I was thinking to update the navidrome package too, can I simply add a commit to this pull request or I need to create a new one?